### PR TITLE
[feat][SFT] Add native Fireworks training backend

### PR DESCRIPTION
**Pull request by bharatmekala**
## What does this PR do?

Adds native supervised fine-tuning on dedicated Fireworks trainers while reusing the existing `SFTTrainer` loop.

- runs trainer-only Fireworks SFT without Ray or a rollout deployment
- reuses native data loading, batching, callbacks, logging, checkpoint cadence, and stateful dataloader resume
- saves resumable DCP state and optionally saves a promotable inference snapshot at every checkpoint cadence
- records the exact Fireworks control-plane DCP row after save, even when Fireworks renames the requested checkpoint
- serializes DCP save and identity resolution against a pre-save control-plane baseline so cross-job resume cannot select an unrelated row
- fails closed when a cross-job manifest lacks a resolved DCP identity
- reuses an existing promotable final checkpoint and deletes the trainer only after successful final promotion

## Checkpoint contract

- `resumable=True` corresponds to DCP weights and optimizer state.
- `promotable=True` corresponds to an inference-formatted sampler checkpoint.
- `save_promotable_checkpoints=false` preserves periodic DCP-only behavior.
- DCP and sampler identities are persisted separately.
- A deleted trainer retains checkpoints for the Fireworks 30-day retention window; cross-job resume uses the exact stored DCP row.

## Validation

- [x] 148 hosted Fireworks backend and SFT config/trainer tests
- [x] repository pre-commit hooks pass
- [x] paid Qwen3-4B same-trainer resume and optimizer restore
- [x] paid Qwen3-4B LoRA promotion
- [x] paid Qwen3.5 35B A3B full-FT promotion
- [x] paid Qwen3.6 35B A3B full-FT smoke completed 38 steps with periodic checkpoints

## CI note

The `fireworks` base branch has repository-wide latest-Ruff drift in the separate SkyRL-CPU workflow, and fork GPU jobs lack Anyscale credentials. The scoped code-quality job passes and the hosted Fireworks suite passes locally.

Generated with [Devin](https://devin.ai)

### CODE REVIEW

File: skyrl/backends/fireworks/sft.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        raise ValueError(f"Fireworks SFT field {name!r} has the wrong batch dimension")
+    return value.detach().cpu()
+
+
+def _tokens(sequences: torch.Tensor, attention_mask: torch.Tensor, row_index: int) -> list[int]:
+    present = [bool(value) for value in attention_mask.tolist()]
+    count = sum(present)
+    if present != [False] * (len(present) - count) + [True] * count:
+        raise ValueError(f"attention_mask[{row_index}] must describe contiguous left padding")
+    return [int(token) for token in sequences[attention_mask.bool()].tolist()]
```
[the remark this answers is not in the document]
Comment (bharatmekala) [reply skyrl/backends/fireworks/sft.py]: Fixed in 284eb026. The converter now checks for a True-to-False tensor transition and reuses the boolean tensor for token extraction; the existing non-contiguous-padding regression test still passes.

**Comment by bharatmekala** [comment]
CI note: the current failures occur before this PRs feature tests run:

- `SkyRL-CPU` reports 2,641 repository-wide Ruff findings already present on the `fireworks` base branch.
- `SkyRL-Train-CPU` Black reformats eight pre-existing Fireworks/base files not changed by this PR.
- `SkyRL-GPU` cannot submit from the fork because the Anyscale token is unavailable.

Focused Fireworks/SFT tests, the existing SFT regression subset, formatting on every changed Python file, and the docs production build pass locally; exact commands and live validation are recorded in the PR body.

**Comment by bharatmekala** [comment]
Paid deleted-trainer DCP resume validation passed on Qwen3-4B LoRA:

- source trainer: `skyrl-smoke-dcp-source-a4825fc411`
- source forward/backward + optimizer step succeeded
- DCP resolved to exact row `.../checkpoints/step-1` with type `CHECKPOINT_TYPE_TRAINING_LORA`
- source trainer deleted and reached `JOB_STATE_DELETED`
- new target trainer: `skyrl-smoke-dcp-target-a4825fc411`
- target loaded `cross_job://skyrl-smoke-dcp-source-a4825fc411/step-1` with `load_state_with_optimizer`
- post-restore forward/backward + optimizer step succeeded
- source and target trainers both verified `JOB_STATE_DELETED` in final cleanup

No deployment was created.
